### ISSUE TITLE
Listen for push branch events instead of comments

### ISSUE THREAD

**Issue by g4s8** [amihaiemil/charles-github-ejb#276]
Feature request: I think it can be useful to configure charles bot to listen for push events in master branch and re-index site on new commits instead of using ticket comments.

**Comment by amihaiemil** [comment]
@g4s8 I thought of that too, but realised most of the times it would be a useless operation:

 * how often would a complete reindex be needed?
 * how do we know that text content (searchable content) was changed and not some CSS/JS?
 * how do we know if it's a new page pushed? 
 * we could probably iterate over the commits' changed files, but then again, how do we know if we should look over ``.md`` files, ``.markdown`` files or simply ``.html`` files?

I think when a complete reindex is needed, it's not a problem to just use comments. And when a new page/blog post is indexed, it's useful to have an Issue with the post's title and short description, so the bot will also tweet about it: see [this](https://github.com/amihaiemil/amihaiemil.github.io/issues/50) Issue and the [tweet](https://twitter.com/charlesmikes/status/986489413113790466) about it.

**Comment by g4s8** [comment]
@amihaiemil right, it make sense. But what do you think about using commit commands? Like Travis CI use `[skip CI]` in commit message to skip build. Charles may re-index site if commit message contains `[reindex site]` for instance.

**Comment by amihaiemil** [comment]
@g4s8 Well, it is not a bad idea but, as simple as the functionality may seem, it requires *a lot* of work, mostly refactoring, adding new implementation of the Notifications and Command logic (we would need a Command implementation based on a Commit, for instance) etc. 

And personally, I don't really have time for it, I am focused on docker-java-api now. If you would like to start with this, I will gladly review a PR. But again, it is a lot of work so start small, if you want to. 

Let me know what you think, I could enroll this project in ZC as well.

**Comment by g4s8** [comment]
@amihaiemil sorry, I also doesn't have enough time for it. Maybe you can publish this project to ZC board and someone will fix it one day?

**Comment by amihaiemil** [comment]
@g4s8 I understand, no problem. I won't put it in ZC, though, at least at the moment, since I myself don't have time for it (I am currently working on that Docker client and afterwards ``comdor``). 

But yeah, maybe later we'll refactor it and implement this functionality, at least just for the fun of it (as I said, it's a lot of redesign to be done).
